src/parallel/stderr.rs

```diff
@@ -0,0 +1,84 @@
+/// Helpers functions for [ChildStderr].
+use std::{io::Read, process::ChildStderr};
+
+use crate::Error;
+
+#[cfg(all(not(unix), not(windows)))]
+compile_error!("Only unix and windows support non-blocking pipes! For other OSes, disable the parallel feature.");
+
+#[cfg(unix)]
+fn get_flags(fd: &std::os::unix::io::RawFd) -> Result<libc::c_int, Error> {
+    let flags = unsafe { libc::fcntl(*fd, libc::F_GETFL, 0) };
+    if flags < 0 {
+        return Err(Error::new(
+            crate::ErrorKind::IOError,
+            format!(
+                "Failed to get flags for child stderr: {}",
+                std::io::Error::last_os_error()
+            ),
+        ));
+    }
+    Ok(flags)
+}
+
+#[allow(unused_variables)]
+pub fn set_non_blocking(stderr: &mut ChildStderr) -> Result<(), Error> {
+    // On Unix, switch the pipe to non-blocking mode.
+    // On Windows, we have a different way to be non-blocking.
+    #[cfg(unix)]
+    {
+        use std::os::unix::io::AsRawFd;
+        let fd = stderr.as_raw_fd();
+        let flags = get_flags(&fd)?;
+
+        if unsafe { libc::fcntl(fd, libc::F_SETFL, flags | libc::O_NONBLOCK) } != 0 {
```

Comment:
Honestly I think you don't need get_flags().

From the [man page of fcntl](https://man7.org/linux/man-pages/man2/fcntl.2.html)

>  On Linux, this command can change
>               only the O_APPEND, O_ASYNC, O_DIRECT, O_NOATIME, and
>               O_NONBLOCK flags.  It is not possible to change the
>               O_DSYNC and O_SYNC flags; see BUGS, below.

It's a pipe on unix, so only O_NONBLOCK makes sense IMHO.

Reply:
Fair enough - I've added debug asserts just to be safe.